src/main/java/com/leverx/mediator/repository/resttemplate/impl/UserRestTemplateRepositoryImpl.java

```diff
@@ -28,9 +28,9 @@
 @Repository
 @Slf4j
 @RequiredArgsConstructor
-public class UserRepositoryImpl implements UserRepository {
+public class UserRestTemplateRepositoryImpl implements UserRepository {
 
-  @Value("${leverx.com.link.sap.users}")
+  @Value("${sap.link.users}")
   private final String usersLink;
 
   private final Auth auth;
```

Comment:
You can register RestTemplate as spring bean.
More info: https://stackoverflow.com/questions/22989500/is-resttemplate-thread-safe